### PR TITLE
Add VK_KHR_maintenance6

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/7172

Most this PR is adding the new `vkCmd*2KHR` versions of the functions

### CODE REVIEW

File: layers/stateless/sl_descriptor.cpp [hunk 1]
```diff
@@ -664,8 +665,10 @@ bool StatelessValidation::ValidateWriteDescriptorSet(const Location &loc, const
             (descriptor_type == VK_DESCRIPTOR_TYPE_SAMPLED_IMAGE) || (descriptor_type == VK_DESCRIPTOR_TYPE_STORAGE_IMAGE) ||
             (descriptor_type == VK_DESCRIPTOR_TYPE_INPUT_ATTACHMENT)) {
             if (descriptor_writes.pImageInfo == nullptr) {
-                const char *vuid = isPushDescriptor ? "VUID-vkCmdPushDescriptorSetKHR-pDescriptorWrites-06494"
-                                                    : "VUID-vkUpdateDescriptorSets-pDescriptorWrites-06493";
+                const char *vuid =
+                    loc.function == Func::vkCmdPushDescriptorSetKHR     ? "VUID-vkCmdPushDescriptorSetKHR-pDescriptorWrites-06494"
+                    : (loc.function == Func::vkCmdPushDescriptorSetKHR) ? "VUID-VkPushDescriptorSetInfoKHR-pDescriptorWrites-06494"
```
Comment (arno-lunarg) [inline layers/stateless/sl_descriptor.cpp:670]:
```suggestion
                    : (loc.function == Func::vkCmdPushDescriptorSet2KHR) ? "VUID-VkPushDescriptorSetInfoKHR-pDescriptorWrites-06494"
```

File: tests/unit/image.cpp [hunk 2]
```diff
@@ -4434,6 +4434,52 @@ TEST_F(NegativeImage, BlockTexelViewLevelOrLayerCount) {
...
+    image.Init(image_create_info, 0);
+
+    VkImageViewCreateInfo ivci = vku::InitStructHelper();
+    ivci.image = image.handle();
+    ivci.viewType = VK_IMAGE_VIEW_TYPE_2D_ARRAY;
+    ivci.format = VK_FORMAT_R16G16B16A16_UNORM;
+    ivci.subresourceRange.baseMipLevel = 0;
+    ivci.subresourceRange.baseArrayLayer = 0;
+    ivci.subresourceRange.aspectMask = VK_IMAGE_ASPECT_COLOR_BIT;
+    ivci.subresourceRange.levelCount = 1;
```
Comment (arno-lunarg) [inline tests/unit/image.cpp:4478]: 07072:
If image was created with the VK_IMAGE_CREATE_BLOCK_TEXEL_VIEW_COMPATIBLE_BIT flag and format is a non-compressed format, the levelCount member of subresourceRange must be 1

here `levelCount` is 1 and the VU is triggered? Maybe am I just missing something

**Review by arno-lunarg — APPROVED** [review]
LGTM
Maybe add a description to the new tests? As mainly see this as a thing for reviewers to quickly understand what is being tested

File: tests/unit/image.cpp (hunk 2 above)
Comment (spencer-lunarg) [reply tests/unit/image.cpp:4478]: yep, should have been `VUID-VkImageViewCreateInfo-image-09487` (the new VU) good catch

File: layers/layer_chassis_dispatch_manual.cpp [hunk 3]
```diff
@@ -872,6 +872,27 @@ void DispatchCmdPushDescriptorSetWithTemplateKHR(VkCommandBuffer commandBuffer,
...
+                                                                                      pPushDescriptorSetWithTemplateInfo);
+    uint64_t template_handle = CastToUint64(pPushDescriptorSetWithTemplateInfo->descriptorUpdateTemplate);
+    void *unwrapped_buffer = nullptr;
+    {
+        ReadLockGuard lock(dispatch_lock);
+        const_cast<VkPushDescriptorSetWithTemplateInfoKHR *>(pPushDescriptorSetWithTemplateInfo)->descriptorUpdateTemplate =
+            layer_data->Unwrap(pPushDescriptorSetWithTemplateInfo->descriptorUpdateTemplate);
+        const_cast<VkPushDescriptorSetWithTemplateInfoKHR *>(pPushDescriptorSetWithTemplateInfo)->layout =
+            layer_data->Unwrap(pPushDescriptorSetWithTemplateInfo->layout);
+        unwrapped_buffer =
```
[the remark this answers is not in the document]
Comment (spencer-lunarg) [reply layers/layer_chassis_dispatch_manual.cpp:889]: wow, nice catch, I forgot to use it with

`const_cast<VkPushDescriptorSetWithTemplateInfoKHR *>(pPushDescriptorSetWithTemplateInfo)->pData = unwrapped_buffer;`

**Pull request merged**